.github/workflows/render-readme.yaml

```diff
@@ -28,17 +28,34 @@ jobs:
       - name: Install dependencies
         uses: r-lib/actions/setup-r-dependencies@v2
         with:
-          extra-packages: any::rmarkdown, local::.
+          extra-packages: any::rmarkdown, local::., any::allcontributors
 
+      - name: Update contributors
+        if: github.ref == 'refs/heads/main'
+        run: allcontributors::add_contributors(format = "text")
+        shell: Rscript {0}
+        
       - name: Compile the readme
         run: |
           rmarkdown::render("README.Rmd")
         shell: Rscript {0}
 
-      - name: Commit files
-        run: |
-          git config --local user.email "action@github.com"
-          git config --local user.name "GitHub Action"
-          git add README.md man/figures/
-          git diff-index --quiet HEAD || git commit -m "Automatic readme update [ci skip]"
-          git push origin || echo "No changes to push"
+      - name: Check for README.Rmd changes
+        id: readme_changes
+        uses: dorny/paths-filter@v3
+        with:
+          filters: |
+            readme:
+              - 'README.Rmd'
```

Comment:
I don't believe this step is necessary as this is already handled in `peter-evans/create-pull-request` from what I can see.

Reply:
Ah good point I've removed.